components/automate-cli/cmd/chef-automate/summary.go

```diff
@@ -0,0 +1,429 @@
+	if len(ss.feStatus) != 0 {
+		fmt.Println("Frontend Services")
+		t := table.NewWriter()
+		tTemp := table.Table{}
+		tTemp.Render()
+		for _, status := range ss.feStatus {
+			t.AppendRow(table.Row{status.serviceName, status.ipAddress, status.status, status.Opensearch})
+		}
+		t.AppendHeader(table.Row{"Name", "IP Address", "Status", "Opensearch"})
+		return t.Render()
+	}
+	return ""
+}
+
+// Display Backend Status
+func (ss *Summary) ShowBEStatus() string {
+	if len(ss.beStatus) != 0 {
+		fmt.Println("Backend Services")
+		t := table.NewWriter()
+		tTemp := table.Table{}
+		tTemp.Render()
+		for _, status := range ss.beStatus {
+			t.AppendRow(table.Row{status.serviceName, status.ipAddress, status.health, status.process, status.upTime, status.role})
+		}
+		t.AppendHeader(table.Row{"Name", "IP Address", "Health", "Process", "Uptime", "Role"})
+		return t.Render()
+	}
+	return ""
+}
+
+// Run Status summary
+func (ss *Summary) Run(sshUtil SSHUtil) error {
+	automateIps, chefServerIps, opensearchIps, postgresqlIps, errList := ss.checkIPAddresses()
+	if errList != nil && errList.Len() > 0 {
+		return status.Wrap(getSingleErrorFromList(errList), status.ConfigError, "IP address validation failed")
+	}
+	err := ss.prepareFEScript(sshUtil, automateIps, "Automate", "FE")
+	if err != nil {
+		return err
+	}
+	err = ss.prepareFEScript(sshUtil, chefServerIps, "Chef Server", "FE")
+	if err != nil {
+		return err
+	}
+	err = ss.prepareBEScript(sshUtil, opensearchIps, "Open Search", "BE")
+	if err != nil {
+		return err
+	}
+	err = ss.prepareBEScript(sshUtil, postgresqlIps, "Postgresql", "BE")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// get sshConfig
+func (ss *Summary) setSSHConfig() SSHUtil {
+	sshconfig := &SSHConfig{}
+	sshconfig.sshUser = ss.infra.Outputs.SSHUser.Value
+	sshconfig.sshKeyFile = ss.infra.Outputs.SSHKeyFile.Value
+	sshconfig.sshPort = ss.infra.Outputs.SSHPort.Value
+	sshUtil := NewSSHUtil(sshconfig)
+	return sshUtil
+}
+
+func (ss *Summary) prepareBEScript(sshUtil SSHUtil, serviceIps []string, serviceName, serviceType string) error {
+	script := ""
+	err := ss.executeCommandForArrayofIPs(sshUtil, serviceIps, script, serviceName, serviceType)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ss *Summary) prepareFEScript(sshUtil SSHUtil, serviceIps []string, serviceName, serviceType string) error {
+	script := GenerateOriginalAutomateCLICommand(
+		&cobra.Command{
+			Use: "chef-automate",
+		}, []string{
+			"status",
+			"-t",
+			"150",
+		})
+
+	err := ss.executeCommandForArrayofIPs(sshUtil, serviceIps, script, serviceName, serviceType)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ss *Summary) executeCommandForArrayofIPs(sshUtil SSHUtil, remoteIps []string, script, serviceName, serviceType string) error {
+	getConfigJsonString := convTfvarToJson(a2haHabitatAutoTfvars)
+	for i := 0; i < len(remoteIps); i++ {
+		sshUtil.getSSHConfig().hostIP = remoteIps[i]
```

Comment:
can we make async ssh calls for executing commands in a loop?

Reply:
While executing commands we are using go routine